exporter/datadogexporter/translate_traces.go

```diff
@@ -549,17 +549,33 @@ func getSpanErrorAndSetTags(s pdata.Span, tags map[string]string) int32 {
 
 	if isError == errorCode {
 		extractErrorTagsFromEvents(s, tags)
-		// If we weren't able to pull an error type or message, go ahead and set
-		// these to the old defaults
-		if _, ok := tags[ext.ErrorType]; !ok {
-			tags[ext.ErrorType] = "ERR_CODE_" + strconv.FormatInt(int64(status.Code()), 10)
-		}
 
+		// if the error message doesnt exist, try to infer useful error information from
+		// the status message, and http status code / http status text.
+		// if we find useful error.message info, also set a fallback error.type
+		// otherwise leave these tags unset and allow the UI to handle defaults
 		if _, ok := tags[ext.ErrorMsg]; !ok {
 			if status.Message() != "" {
 				tags[ext.ErrorMsg] = status.Message()
+
+				// If we weren't able to pull an error type, set to a reasonable default
+				if _, ok := tags[ext.ErrorType]; !ok {
+					tags[ext.ErrorType] = "error"
+				}
 			} else {
-				tags[ext.ErrorMsg] = "ERR_CODE_" + strconv.FormatInt(int64(status.Code()), 10)
+				// look for useful http metadata if it exists and add that as a fallback for the error message
+				if statusCode, ok := tags[conventions.AttributeHTTPStatusCode]; ok {
```

Comment:
After the other comment is addressed you can collapse this into an `else if`

Reply:
i collapsed this a bit to remove 1 layer of nested ifs, lmk if u had a diff idea?

Reply:
That's what I had in mind, IMO it looks much better now!